draft-ietf-tls-esni.md

```diff
@@ -699,24 +699,30 @@ MAY offer to resume sessions established without ECH.
 ## Client-Facing Server
 
 Upon receiving an "encrypted_client_hello" extension, the client-facing server
-MUST check that it is able to negotiate TLS 1.3 or greater. If not, it MUST
-abort the connection with a "handshake_failure" alert.
+first checks that it is able to negotiate TLS 1.3 or higher. If not, then it
+MUST ignore the extension and proceed with the handshake as usual. Next, the
```

Comment:
Hrm. Actually this whole bit is kind of odd now, because the inner and outer ClientHellos no longer have the same version list. I think it's something that wasn't updated in the ESNI to ECH change.

FWIW, I don't think it needed changes for GREASE. A client greasing ECH presumably supports TLS 1.3, and any server that implements this specification would also support TLS 1.3. But it might need some tweaks on account of ECH. It seems the version check should happen after you've chosen to use the inner ClientHello, and ECH wants to get resolved extremely early, before you've actually locked in a version.

Reply:
What's your proposal? If the we add a rule that says the client MUST NOT (or at least SHOULD NOT) offer GREASE in TLS 1.2 or below, then I'm happy to revert to aborting the handshake here.

Reply:
I'll upload a PR for this. I don't think either the old or new text is right. (See https://github.com/tlswg/draft-ietf-tls-esni/issues/309#issuecomment-702233855)

Reply:
https://github.com/tlswg/draft-ietf-tls-esni/pull/320